### PR TITLE
nixos/snapserver: update module to work with snapcast 0.20

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change

With the update of snapcast from 0.15 to 0.20 in https://github.com/NixOS/nixpkgs/pull/90294 I inadvertently broke the module, since the CLI parameters changed quite a bit.

This should rectify and even improve the situation.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
For some reason I'm always gettingthe renamed trace, even though I'm not using the controlPort variable any more. This is also visible in the tests.

```
trace: Obsolete option `services.snapserver.controlPort' is used. It was renamed to `services.snapserver.tcp.port'.
```

**Comment by tobim** [comment]
I'll try to squeeze a review in later today.

**Comment by mweinelt** [comment]
@tobim I also believe we need to add a static group for snapcast, since we need to allow other users like `mpd` to write to the pipe at `/run/snapserver/mpd`.

File: nixos/modules/services/audio/snapserver.nix [hunk 1]
```diff
@@ -31,27 +31,42 @@ let
...
+     # http json rpc
+    ++ [ "--http.enabled ${toString cfg.http.enable}" ]
+    ++ optionals cfg.http.enable [
+      "--http.address ${cfg.http.listenAddress}"
+      "--http.port ${toString cfg.http.port}"
+    ] ++ optional (cfg.http.docRoot != null) "--http.doc_root \"${toString cfg.http.docRoot}\"");
 
 in {
+  imports = [
+    (mkRenamedOptionModule [ "services" "snapserver" "controlPort"] [ "services" "snapserver" "tcp" "port" ])
```
Comment (tobim) [inline nixos/modules/services/audio/snapserver.nix:68]: Adjust line 282 to fix the deprecation warning.

**Review by tobim — CHANGES_REQUESTED** [review]
> I also believe we need to add a static group for snapcast, since we need to allow other users like `mpd` to write to the pipe at `/run/snapserver/mpd`.

The server always creates the pipe with write access for all:
https://github.com/badaix/snapcast/blob/d44f3be4edee76494cecb1c28caf46969f3ba15c/server/streamreader/pipe_stream.cpp#L50

If we need access control for that it should be implemented upsteam.

File: nixos/modules/services/audio/snapserver.nix (hunk 1 above)
Comment (mweinelt) [reply nixos/modules/services/audio/snapserver.nix:68]: Fixed. Good catch.

**Comment by mweinelt** [comment]
Indeed, thanks for the pointer to the snapcast source code.

**Comment by mweinelt** [comment]
@GrahamcOfBorg test snapcast

**Review by tobim — APPROVED** [review]
Tested locally, works with `mpd`.

**Comment by mweinelt** [comment]
Thanks for reviewing this!

**Pull request merged**